### PR TITLE
Activity Log: Use Activity Log v2 in all versions of Calypso

### PR DESCRIPTION
**Pull request by mattgawarecki**
-----

***NOTE:** Before this PR is merged, we'll need to add an illustration for the Activity Log's "empty" state. I've contacted our Design team for help coming up with one, but for now I've added a placeholder that just displays the string "Empty!" I'm happy to allow review, but I'll wait to deploy until the empty state is fully polished.*

-----

Since Jetpack Cloud went live, we've used two different versions of the Activity Log: the existing version for Calypso Blue, and a slight re-imagining of it in Calypso Green. This PR is a proposal for us to eliminate the older Calypso Blue version and replace it with the one currently used in Calypso Green. As a result, this PR would:

* reduce the surface area for bugs;
* make Calypso's Activity Log easier to maintain;
* unify the Activity Log experience between the two Calypsos; and
* significantly reduce the size of the bundle where the code for the Activity Log lives, improving performance on any page that references it.

##### The big stuff

* Modify `ActivityCardList` so that it always uses `ActivityCard` instead of `ActivityLogItem`, removing any related feature flag checks.
* Account for the "empty" state (i.e., when a site has no activities logged) by showing a relevant message on the Activity Log page.
* Move the "v1" Activity Log components that are still used in a specific scope, so that they're inside that scope in the file hierarchy: `ActivityLogItem`, `ActivityLogConfirmDialog`, and `ActivityLogRewindToggle`.

##### Janitorial stuff that follows as a result of the big stuff

* Remove the `activity-log/v2` feature flag from all config files.
* Remove all un-referenced "v1" Activity Log components: `ActivityLog`, `ActivityLogBanner`, `ActivityLogSwitch`, `ActivityLogTaskList`, and others.
* Remove the `ActivityLogExample` component, since `ActivityLogItem` is no longer meant to be widely used.

#### Testing instructions

1. Check out this PR (`git checkout remove/old-activity-log`) and start Calypso Blue (`yarn start`).
2. Select any Jetpack site. Verify the Activity Log works exactly as it does on Jetpack Cloud now, with no loss of functionality over the "original" Calypso Blue Activity Log.
3. Select any WordPress.com-hosted site (Atomic, Simple, etc.). Verify the Activity Log works for these sites as well, with no loss of functionality over the existing Calypso Blue Activity Log implementation.
4. Select a site with no logged activities (e.g., a site in "Coming soon" mode; a domain-only site; a site that's existed without updates for long enough that we would no longer show those activities on the page).

### CODE REVIEW

**Comment by elliottprogrammer** [comment]
@mattgawarecki, One of the first things I'm noticing is that in the new Activity log, there are various banners and upsells missing, compared to the old Calypso blue Activity log: (see screenshots)

These banners contain the ability to upgrade and to update the site's plugins, so I'm thinking we would probably want these banners to remain in the (Calypso blue) Activity Log, right?  What do you think?

Markup 2022-05-03 at 17 32 31

Markup 2022-05-03 at 17 33 31

File: client/components/activity-card-list/index.jsx [hunk 1]
```diff
@@ -209,6 +209,10 @@ class ActivityCardList extends Component {
 		const pageLogs = this.splitLogsByDate( visibleLogs.slice( ( actualPage - 1 ) * pageSize ) );
 		const showLimitUpsell = visibleLogs.length < logs.length && actualPage >= pageCount;
 
+		if ( pageLogs.length === 0 ) {
+			return 'Empty!';
```
Comment (manzoorwanijk) [inline client/components/activity-card-list/index.jsx:213]: Doesn't this need to be translated? I see many other Pagination labels in this component that have not been translated, any specific reason? 🤔

File: client/my-sites/activity/activity-log-v2/index.tsx [hunk 2]
```diff
@@ -95,9 +95,10 @@ const ActivityLogV2: FunctionComponent = () => {
 				jetpackCloudHeader
 			) : (
 				<FormattedHeader
+					brandFont
 					headerText="Activity"
 					subHeaderText={ translate(
-						'This is the complete event history for your site. Filter by date range and/or activity type.'
+						"Keep tabs on all your site's activity — plugin and theme updates, user logins, setting modifications, and more."
```
Comment (manzoorwanijk) [inline client/my-sites/activity/activity-log-v2/index.tsx:101]: Should this be `setting modifications` or `settings modifications`?

diff
```diff
- setting modifications
+ settings modifications
```

File: client/components/activity-card-list/index.jsx (hunk 1 above)
Comment (mattgawarecki) [reply client/components/activity-card-list/index.jsx:213]: It definitely would, if we ever wanted to put this into production. Fortunately in this case, we'll eventually have an illustration (and maybe a little copy) to replace the `Empty!` string.

Comment (mattgawarecki) [reply client/components/activity-card-list/index.jsx:213]: As for the other labels, those need to be translated as well, but I tried to keep my first draft of changes as finely scoped as possible.

File: client/my-sites/activity/activity-log-v2/index.tsx (hunk 2 above)
Comment (mattgawarecki) [reply client/my-sites/activity/activity-log-v2/index.tsx:101]: I'm no copy expert, but I think both forms could work. Personally my brain gravitates more toward the plural "settings modifications," but I think either could be grammatically correct. It's worth noting, this copy is pulled directly from the original Calypso Blue `ActivityLog` component, so it's already got translations on file.

**Comment by mattgawarecki** [comment]
> One of the first things I'm noticing is that in the new Activity log, there are various banners and upsells missing, compared to the old Calypso blue Activity log

@elliottprogrammer You are absolutely right! I completely missed a couple page elements that are present in Calypso Blue but not Green 🤦  I'll look into how big of a deal it would be to port these over. Depending on what I find, it might be easier to abandon this PR for a later date when we can give it more focus.

> the ability to upgrade and to update the site's plugins

We have an "upgrade" element that we use in Calypso Green, but it only shows up when the user could benefit from upgrading (e.g., if they have a Free plan and more than 20 activities logged). I wonder if the Green version would work here, as well?

The ability to see which plugins need updates is definitely not part of the "v2" Activity Log, though, which I think is something worth addressing -- either here, or in another PR.

File: client/components/activity-card-list/index.jsx (hunk 1 above)
Comment (manzoorwanijk) [reply client/components/activity-card-list/index.jsx:213]: May be we can track that somewhere so that it doesn't get unnoticed?

Comment (mattgawarecki) [reply client/components/activity-card-list/index.jsx:213]: For now I've added a note to the top of the PR description and added the `DO NOT MERGE` label -- does that match up with what you're thinking, or is there something else we can do as well?

Comment (manzoorwanijk) [reply client/components/activity-card-list/index.jsx:213]: That "`DO NOT MERGE` label" works fine. Thank you.

**Comment by keoshi** [comment]
@mattgawarecki Removing Jetpack Design from this PR for the time being based on the [last comment here](https://github.com/Automattic/wp-calypso/pull/63223#issuecomment-1118996024). If you'd still like us to weigh in here just holla at us, pleas!

**Pull request closed without merging**